gensim/models/word2vec.py

```diff
@@ -654,27 +654,27 @@ def finalize_vocab(self):
             # create null pseudo-word for padding when using concatenative L1 (run-of-words)
             # this word is only ever input – never predicted – so count, huffman-point, etc doesn't matter
             word, v = '\0', Vocab(count=1, sample_int=0)
-            v.index = len(self.vocab)
-            self.index2word.append(word)
-            self.vocab[word] = v
+            v.index = len(self.kv.vocab)
+            self.kv.index2word.append(word)
+            self.kv.vocab[word] = v
         # set initial input/projection and hidden weights
         self.reset_weights()
 
     def sort_vocab(self):
         """Sort the vocabulary so the most frequent words have the lowest indexes."""
-        if hasattr(self, 'syn0'):
+        if self.kv.syn0:
```

Comment:
@jayantj `self.kv.syn0` gets initialized in `__init__` but isn't populated, so if the vectors/weights have not been initialized it will remain an empty list and an empty list as a conditional is `False